### PR TITLE
lesson 06 mock tests

### PR DESCRIPTION
**Pull request by nubber7**
test_mailroom_v4.py::test_thank_you_letter_positive[dan-50-Thank you dan for donating 50 dollars generously.] PASSED                                                                   [ 10%]
test_mailroom_v4.py::test_thank_you_letter_positive[jeff-60-Thank you jeff for donating 60 dollars generously.] PASSED                                                                 [ 20%]
test_mailroom_v4.py::test_thank_you_letter_negitive[dan-50-Thank you sam for donating 50 dollars generously.] PASSED                                                                   [ 30%]
test_mailroom_v4.py::test_thank_you_letter_negitive[jeff-60-Thank you  for donating 60 dollars generously.] PASSED                                                                     [ 40%]
test_mailroom_v4.py::test_donor_details PASSED                                                                                                                                         [ 50%]
test_mailroom_v4.py::test_amount_validate_positive PASSED                                                                                                                              [ 60%]
test_mailroom_v4.py::test_amount_validate_negitive PASSED                                                                                                                              [ 70%]
test_mailroom_v4.py::test_thank_you PASSED                                                                                                                                             [ 80%]
test_mailroom_v4.py::test_create_report PASSED                                                                                                                                         [ 90%]
test_mailroom_v4.py::test_send_letters_all_call PASSED                                                                                                                                 [100%]

### CODE REVIEW

File: students/g_rama/lesson06/test_mailroom_v4.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    mock_donor_details.assert_called_with(testing_donors_data)
+
+
+def test_amount_validate_positive():
+    assert mailroom_v4.amount_validate(float(20))
+
+
+@unittest.mock.patch('mailroom_v4.amount_validate')
+def test_amount_validate_negitive(mock_amount_validate):
+    mailroom_v4.amount_validate(-10)
```
Comment (natasha-aleksandrova) [inline students/g_rama/lesson06/test_mailroom_v4.py]: same comment here - you're testing the mock.

File: students/g_rama/lesson06/test_mailroom_v4.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                   "testname2": [500, 20],
+                   "Susan": [1000, 20, 70],
+                   "Rob": [250, 20],
+                   }
+
+
+@unittest.mock.patch('mailroom_v4.donor_details')
+def test_donor_details(mock_donor_details):
+    mailroom_v4.donor_details(testing_donors_data)
+    mock_donor_details.assert_called_with(testing_donors_data)
```
Comment (natasha-aleksandrova) [inline students/g_rama/lesson06/test_mailroom_v4.py]: same comment here - you're testing the  mock.

File: students/g_rama/lesson06/mailroom_v4.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+               "Jeff": [500, 20],
+               "Susan": [1000, 20, 70],
+               "Rob": [250, 20],
+               "Ross": [200]}
+
+
+def display_donors():
+    """Display donors if user enters list"""
+    for _ in donors_data.keys():
+        print('{}'.format(_))
```
[the remark this answers is not in the document]
Comment (gunturur) [reply students/g_rama/lesson06/mailroom_v4.py]: Got it. .Didn't see this comment. Will not use _ variable

File: students/g_rama/lesson06/test_mailroom_v4.py (hunk 1 above)
Comment (gunturur) [reply students/g_rama/lesson06/test_mailroom_v4.py]: Kind of confused when you say "testing the mock" Although function is mocked, I am calling the  function from mailroom_v4? doesn't it mean I will be testing the functionality of that function?

**Pull request merged**